src/codeu/chat/client/core/Context.java

```diff
@@ -1,60 +1,33 @@
-// Copyright 2017 Google Inc.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//    http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package codeu.chat.client.core;
 
 import java.util.ArrayList;
 import java.util.Collection;
 
-import codeu.chat.common.BasicView;
 import codeu.chat.common.User;
-import codeu.chat.common.VersionInfo;
+import codeu.chat.client.core.UserContext;
 import codeu.chat.util.connections.ConnectionSource;
-import codeu.chat.util.ServerInfo;
-import codeu.chat.util.Uuid;
-
-public final class Context {
 
-  private final BasicView view;
-  private final Controller controller;
+public class Context extends codeu.chat.contexts.Context {
 
   public Context(ConnectionSource source) {
-    this.view = new View(source);
-    this.controller = new Controller(source);
+    super(new View(source), new Controller(source));
   }
 
+  @Override
   public UserContext create(String name) {
     final User user = controller.newUser(name);
     return user == null ?
         null :
-        new UserContext(user, view, controller);
+        new UserContext(user, (View) view, (Controller) controller);
```

Comment:
ah, ok. i think what threw me off is that instinctively i'd expect basicview/basiccontroller to be subclasses, not superclasses, of view and controller. nevermind then.